src/main/java/com/gotocompany/depot/maxcompute/converter/ConverterOrchestrator.java

```diff
@@ -0,0 +1,77 @@
+package com.gotocompany.depot.maxcompute.converter;
+
+import com.aliyun.odps.type.TypeInfo;
+import com.google.protobuf.Descriptors;
+import com.gotocompany.depot.config.MaxComputeSinkConfig;
+import com.gotocompany.depot.maxcompute.converter.payload.DurationPayloadConverter;
+import com.gotocompany.depot.maxcompute.converter.payload.MessagePayloadConverter;
+import com.gotocompany.depot.maxcompute.converter.payload.PayloadConverter;
+import com.gotocompany.depot.maxcompute.converter.payload.PrimitivePayloadConverter;
+import com.gotocompany.depot.maxcompute.converter.payload.StructPayloadConverter;
+import com.gotocompany.depot.maxcompute.converter.payload.TimestampPayloadConverter;
+import com.gotocompany.depot.maxcompute.converter.type.DurationTypeInfoConverter;
+import com.gotocompany.depot.maxcompute.converter.type.MessageTypeInfoConverter;
+import com.gotocompany.depot.maxcompute.converter.type.PrimitiveTypeInfoConverter;
+import com.gotocompany.depot.maxcompute.converter.type.StructTypeInfoConverter;
+import com.gotocompany.depot.maxcompute.converter.type.TimestampTypeInfoConverter;
+import com.gotocompany.depot.maxcompute.converter.type.TypeInfoConverter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ConverterOrchestrator {
+
+    private final List<TypeInfoConverter> typeInfoConverters;
+    private final List<PayloadConverter> payloadConverters;
+    private final Map<String, TypeInfo> typeInfoCache;
+
+    public ConverterOrchestrator(MaxComputeSinkConfig maxComputeSinkConfig) {
+        typeInfoConverters = new ArrayList<>();
+        payloadConverters = new ArrayList<>();
+        typeInfoCache = new ConcurrentHashMap<>();
+        initializeConverters(maxComputeSinkConfig);
+    }
+
+    public TypeInfo convert(Descriptors.FieldDescriptor fieldDescriptor) {
+        return typeInfoCache.computeIfAbsent(fieldDescriptor.getFullName(), key -> typeInfoConverters.stream()
+                .filter(converter -> converter.canConvert(fieldDescriptor))
+                .findFirst()
+                .map(converter -> converter.convert(fieldDescriptor))
+                .orElseThrow(() -> new IllegalArgumentException("Unsupported type: " + fieldDescriptor.getType())));
+    }
```

Comment:
The converter selection logic uses stream operations which create unnecessary intermediate objects. Consider using a Map<Type, Converter> for direct lookups which would improve performance and reduce garbage collection overhead.

Reply:
Defer to the next phase

Reply:
ToDo phase 2